pkg/sql/stats/automatic_stats.go

```diff
@@ -472,8 +473,6 @@ func (r *Refresher) Start(
 							break
 						case <-r.drainAutoStats:
 							return
-						case <-stopper.ShouldQuiesce():
-							return
```

Comment:
If you are going remove the stopper.ShouldQuiesce() I would add a `<-ctx.Done()` branch to ensure we are making an equivalent change. Same in the other two locations.